src/map/entities/charentity.cpp

```diff
@@ -439,40 +439,34 @@ void CCharEntity::pushPacket(std::unique_ptr<CBasicPacket>&& packet)
 void CCharEntity::updateCharPacket(CCharEntity* PChar, ENTITYUPDATE type, uint8 updatemask)
 {
     auto existing = PendingCharPackets.find(PChar->id);
-    if (existing == PendingCharPackets.end())
+    if (existing == PendingCharPackets.end() || !existing->second)
```

Comment:
I find this logical block quite hard to reason about, can you break these up even if it ends up with nested conditions?

I think it would also help to rename existing to itr, and break out conditions like:

`const bool hasPendingPacket = itr != PendingCharPackets.end();`

and to not use the itr pair directly, but to use:

`auto& pendingPacket = existing->second;`
`pendingPacket ->updateWith...`

Reply:
Updated the logic here, hopefully more clear.